Tests/StarknetTests/Data/JsonRpcResponseTests.swift

```diff
@@ -56,4 +60,49 @@ final class JsonRpcResponseTests: XCTestCase {
         XCTAssertNotNil(response.error)
         XCTAssertNotNil(response.error!.data)
     }
+
+    func testErrorWithStringData() async throws {
+        let json = """
+        {
+            "id": 0,
+            "jsonrpc": "2.0",
+            "error": {
+                "code": 40,
+                "message": "Contract error",
+                "data": "More data about the execution failure."
+            }
+        }
+        """.data(using: .utf8)!
+
+        let decoder = JSONDecoder()
+
+        let response = try decoder.decode(JsonRpcResponse<Int>.self, from: json)
+        XCTAssertNil(response.result)
+        XCTAssertNotNil(response.error)
+        XCTAssertNotNil(response.error!.data)
```

Comment:
Any reason to not assert `code`, `message` and `data` specifically? We know we're decoding but there's no guarantees it is decoding the expected values. Seems relevant in particular for `data`.

Reply:
> Seems relevant in particular for data.

I agree, as data is not being decoded the standard way, it's better to additionally test it. 

I don't think, however, that we should be testing every component of swift standard library. That would significantly bloat the test cases, while not providing much value.